internal/service/datazone/form_type.go

```diff
@@ -0,0 +1,317 @@
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	in := &datazone.CreateFormTypeInput{}
+	in.DomainIdentifier = plan.DomainIdentifier.ValueStringPointer()
+	resp.Diagnostics.Append(flex.Expand(ctx, &plan, in)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	option := flex.WithIgnoredFieldNames([]string{"Model"})
+	out, err := conn.CreateFormType(ctx, in)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.DataZone, create.ErrActionCreating, ResNameFormType, plan.Name.String(), err),
+			err.Error(),
+		)
+		return
+	}
+	if out == nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.DataZone, create.ErrActionCreating, ResNameFormType, plan.Name.String(), nil),
+			errors.New("empty output").Error(),
+		)
+		return
+	}
+
+	createTimeout := r.CreateTimeout(ctx, plan.Timeouts)
+	outputRaws, err := tfresource.RetryWhenNotFound(ctx, createTimeout, func() (interface{}, error) {
+		return findFormTypeByID(ctx, conn, *out.DomainId, *out.Name, *out.Revision)
+	})
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.DataZone, create.ErrActionCreating, ResNameFormType, plan.Name.String(), err),
+			err.Error(),
+		)
+		return
+	}
+	output := outputRaws.(*datazone.GetFormTypeOutput)
+	resp.Diagnostics.Append(flex.Flatten(ctx, output, &plan, option)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	plan.OriginDomainId = flex.StringToFrameworkLegacy(ctx, output.OriginDomainId)
+	plan.OriginProjectId = flex.StringToFrameworkLegacy(ctx, output.OriginProjectId)
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+}
+
+func (r *resourceFormType) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	conn := r.Meta().DataZoneClient(ctx)
+
+	var state resourceFormTypeData
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	out, err := findFormTypeByID(ctx, conn, state.DomainIdentifier.ValueString(), state.Name.ValueString(), state.Revision.ValueString())
+	if tfresource.NotFound(err) {
+		resp.State.RemoveResource(ctx)
+		return
+	}
+	if err != nil {
+		resp.Diagnostics.AddError(
+			create.ProblemStandardMessage(names.DataZone, create.ErrActionSetting, ResNameFormType, state.Name.String(), err),
+			err.Error(),
+		)
+		return
+	}
+	option := flex.WithIgnoredFieldNames([]string{"Model"})
+
+	resp.Diagnostics.Append(flex.Flatten(ctx, out, &state, option)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	state.OwningProjectIdentifier = flex.StringToFramework(ctx, out.OwningProjectId)
+	state.OriginDomainId = flex.StringToFrameworkLegacy(ctx, out.OriginDomainId)
+	state.OriginProjectId = flex.StringToFrameworkLegacy(ctx, out.OriginProjectId)
+
+	resp.Diagnostics.Append(resp.State.Set(ctx, &state)...)
+}
+
+func (r *resourceFormType) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+}
```

Comment:
This can be removed completely since it is embedded with `framework.WithNoUpdate`